[Minor][MLLIB] Refactor toString method in MLLIB

aihex opened the pull request:
1. predict(predict.toString) has already output prefix “predict” thus it’s duplicated to print ", predict = " again
2. there are some extra spaces

Commit:
[Minor][MLLIB] Fix a formatting bug in toString method in Node.scala

1. predict(predict.toString) has already output prefix “predict” thus
it’s
duplicate to print ", predict = " again
2. there are some extra spaces

AmplabJenkins commented:
Can one of the admins verify this patch?

srowen commented on mllib/src/main/scala/org/apache/spark/mllib/tree/model/Node.scala:
```diff
@@ -51,8 +51,8 @@ class Node (
     var stats: Option[InformationGainStats]) extends Serializable with Logging {
 
   override def toString: String = {
-    "id = " + id + ", isLeaf = " + isLeaf + ", predict = " + predict + ", " +
```
These can use string interpolation. I take your point though it breaks the symmetry a bit and make this `toString` rely on details of the subclass. How about making the `Predict.toString` return something more compact like `s"$predict ($prob)"`?

AiHe commented:
Modify Predict.scala and decouple the dependency of Node.scala on Predict.scala.

srowen commented:
I'm OK with this, though while you're at it, why not make these use string interpolation? `s"foo = $foo, ..."`

AiHe commented:
If you're referring to the modification on Predict.scala, I just follow the style of previous code.

srowen commented:
Yes, but that's a hold over from Scala 2.9 and before. I think that while we're modifying these strings they can use the 'modern' style used in new code in the project.

Commit:
[Minor][MLLIB] Fix a formatting bug in toString method in Node

Modify Predict.scala and decouple the dependency of Node.scala on
Predict.scala

AiHe commented:
Okay, sounds better to use a modern style.

srowen commented:
... this still is not using string interpolation everywhere. It's a small thing, but if we're bothering to update these `toString` methods, go all the way.

jkbradley commented on mllib/src/main/scala/org/apache/spark/mllib/tree/model/Predict.scala:
```diff
@@ -29,9 +29,7 @@ class Predict(
-  override def toString: String = {
-    "predict = %f, prob = %f".format(predict, prob)
-  }
+  override def toString: String = s"$predict(prob = $prob)"
```
I'd put a space before the parenthesis: ```$predict (prob```

Commit:
Minor][MLLIB] Format toString method in MLLIB

Change to string interpolation way

AiHe commented:
Get your point and change all toString methods in MLLIB. 
There are a large number of uses of 'old' way in the statement like logInfo, which makes it hard to change all them in a moment.

srowen commented:
Looks OK, I think Joseph just had one more tiny comment on one `toString` method, to add a space.

AiHe commented:
Good to go. 
@jkbradley Thank you.

srowen commented:
ok to test

jkbradley commented:
LGTM pending tests

SparkQA commented:
[Test build #699 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/699/consoleFull) for   PR 5687 at commit [`b6380c6`](https://github.com/apache/spark/commit/b6380c6770a913efe07c0f8fd8621843cced9fc4).

SparkQA commented:
[Test build #699 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/699/consoleFull) for   PR 5687 at commit [`b6380c6`](https://github.com/apache/spark/commit/b6380c6770a913efe07c0f8fd8621843cced9fc4).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch **adds the following new dependencies:**
   * `tachyon-0.6.4.jar`
   * `tachyon-client-0.6.4.jar`

 * This patch **removes the following dependencies:**
   * `tachyon-0.5.0.jar`
   * `tachyon-client-0.5.0.jar`

srowen commented:
@AiHe  this fails style checks. I think the lines are too long. Can you fix those up and try `./dev/lint-scala` to check it out locally?

Commit:
Pass scala coding style checking

AiHe commented:
@srowen Sure. Pass the style checking and will do that for PRs in the feature.

SparkQA commented:
[Test build #710 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/710/consoleFull) for   PR 5687 at commit [`9862b9a`](https://github.com/apache/spark/commit/9862b9a716f1afdff384f5d155846b4c99683b9d).

SparkQA commented:
[Test build #710 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/710/consoleFull) for   PR 5687 at commit [`9862b9a`](https://github.com/apache/spark/commit/9862b9a716f1afdff384f5d155846b4c99683b9d).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch does not change any dependencies.
